server/server/src/sync/remote_data_synchroniser.rs

```diff
@@ -309,6 +306,13 @@ impl<'a> RemoteSyncState<'a> {
     }
 
     pub fn set_initial_remote_data_synced(&self) -> Result<(), RepositoryError> {
+        let cursor = self
+            .changelog_row_repository
+            .latest_changelog()?
+            .map(|row| row.id)
+            .unwrap_or(0) as u32;
+        self.update_push_cursor(cursor + 1)?;
```

Comment:
Think its better to keep the logic of calculating the latest cursor out of this struct impl. Intention of this RemoteSyncState struct was to have a getter/setter interface for the key value store...

Reply:
Hmm, the name of the struct `RemoteSyncState` and the name of the method `set_initial_remote_data_synced` to me suggests it's beyond just key values store, it's for settings/getting sync state.  And I can't see a scenario  where you would want to set `initialisation finished` without updating remote sync cursor, is there one ? Happy to change this but i think when we call something that `sets initialisation finished`, it should do both update the cursor and initialisation state (those two tasks seem coupled)

Reply:
Maybe just move the querying and calculation of the new push cursor on level up so that RemoteSyncState only updates the state in the DB but doesn't has any logic:

```rust
 pub fn set_initial_remote_data_synced(&self, push_cursor: u32) -> Result<(), RepositoryError> 
```

Reply:
Sounds like you really want the `RemoteSyncState` to be logic free, i reverted it an just disabling sync in cli when refresh data or initialise from export 65fae5105244441c7beccb31034fab631cdbdf2c, also to be clear to future developers, added comments: f0694cd4c73faf0667e79a9d6c3ea71fe30fc94d